week3/homework/src/functions.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+const express = require('express');
+
+const uuid = require('uuid/v4');
+
+const app = express();
+
+const {
+  readFile: _readFile,
+  writeFile: _writeFile
+} = require('fs');
+
+const {
+  promisify
+} = require('util');
+
+const readFile = promisify(_readFile);
+const writeFile = promisify(_writeFile);
+
+
+
+//USE BUILT-IN JSON middleware to automatically parse JSON
+app.use(express.json());
+
+const TODO_FILE = 'todo.json';
+
+function readTodos() {
+  return readFile(TODO_FILE, 'utf-8').then(JSON.parse, () => []);
+}
+
+function writeTodos(todos) {
+  return writeFile(TODO_FILE, JSON.stringify(todos));
+}
+
+//Find Todo
+async function findTodo(req) {
+  const id = req.params.id;
+  const todos = await readTodos();
+  const todo = todos.find(todos => todos.id === id);
+  const done = req.body.done;
+  return await [
+    id,
+    req.body,
+    done,
+    todo,
+    todos
+  ];
+}
+
+//MarkTodo Function
+async function markTodo(req, res, doneFlag) {
+  const [, ,
+    done,
+    todo,
+    todos
+  ] = await findTodo(req);
+  if (typeof todo === 'undefined') {
+    return new Error(res.send('NO SUCH ID'));
```

Comment:
Returning a `new Error` here does not make sense to me. Simply return `res.send('NO SUCH ID')`. Makes sense? Also: you should return a json for the errors as well, like so: `{ error: 'NO SUCH ID' }`. It will help us use that error when using this in the browser when we code our front-end application.